src/DGtal/io/viewers/Viewer3D.ih

```diff
@@ -1518,29 +1524,50 @@ void
 DGtal::Viewer3D<Space, KSpace>::glCreateListQuadMaps(const typename DGtal::Display3D<Space, KSpace>::QuadsMap &aQuadMap,
                                                     unsigned int idList){
   glNewList ( idList, GL_COMPILE );
-
-  for (typename DGtal::Display3D<Space, KSpace>::QuadsMap::const_iterator it = aQuadMap.begin(), itE = aQuadMap.end();
-       it != itE; ++it )
+  for (auto & mapElem: aQuadMap)
     {
-      glPushName ( it->first );
+      glPushName ( mapElem.first );
       glEnable ( GL_LIGHTING );
       glBegin ( GL_QUADS );
-
-      for (typename std::vector<typename Viewer3D<Space, KSpace>::QuadD3D>::const_iterator  it_s = it->second.begin();
-           it_s != it->second.end(); it_s++)
+      bool useColorSelection = false;
+      if(mySelectedElementId == mapElem.first)
+        useColorSelection = true;
+      
+      for (auto &q: mapElem.second)
         {
-          glColor4ub ( it_s->color.red(), it_s->color.green(), it_s->color.blue(), it_s->color.alpha() );
-          glNormal3f ( it_s->nx, it_s->ny ,it_s->nz );
-          glVertex3f ( it_s->point1[0], it_s->point1[1], it_s->point1[2] );
-          glVertex3f ( it_s->point2[0], it_s->point2[1], it_s->point2[2] );
-          glVertex3f ( it_s->point3[0], it_s->point3[1], it_s->point3[2] );
-          glVertex3f ( it_s->point4[0], it_s->point4[1], it_s->point4[2] );
+          if(useColorSelection)
+            {
+              unsigned char m = std::max(std::max( q.color.red(), q.color.green()), q.color.blue());
```

Comment:
I think it will give nicer results if you highlight when the *average* of RGB is below a threshold (or make darker in the other case). Here, (0,0,255) (blue) is made darker, which is kind of weird since our eye does not perceive blue very light.